### PR TITLE
Add test vectors for Crypto::Bcrypt.

### PR DESCRIPTION
**Pull request by technorama**
Most are commented out waiting for Bcrypt::Base64 to be fixed.

### CODE REVIEW

File: spec/std/crypto/bcrypt_spec.cr [hunk 1]
```diff
@@ -1,7 +1,39 @@
...
 describe "Bcrypt" do
+  it "tests against other implementations test vectors" do
+    TEST_VECTORS.each do |vec|
+      secret, salt, digest = vec
+      Crypto::Bcrypt.verify(secret, digest)
+    end
+  end
+
+  pending "test password generation with a specific salt against test vectors" do
+  end
```
Comment (jhass) [inline spec/std/crypto/bcrypt_spec.cr]: `pending` is for tests not yet passing, not for tests not yet written ;)

**Comment by asterite** [comment]
@technorama can the commented lines be uncommented? Base64 should be fixed now.

**Comment by jhass** [comment]
I merely added another Base64 API, which isn't even used by the BCrypt implementation. The BCrypt implementation needs some major rework and fixes.

**Comment by asterite** [comment]
Ah, OK, I thought they were related.

**Comment by technorama** [comment]
I uncommented everything and changed the spec to pending.  The specs will be useful to those attempting to fix the bcrypt implementation.

File: spec/std/crypto/bcrypt_spec.cr [hunk 2]
```diff
@@ -1,7 +1,37 @@
...
+  { "\u00A3", "$2a$05$CCCCCCCCCCCCCCCCCCCCC.", "$2a$05$CCCCCCCCCCCCCCCCCCCCC.CAzSxlf0FLW7g1A5q7W/ZCj1xsN6A.e" },
+]
+
+
 describe "Bcrypt" do
+  pending "tests against other implementations test vectors" do
+    TEST_VECTORS.each do |vec|
+      secret, salt, digest = vec
+# BUG: test password generation with salt against test vectors
+      Crypto::Bcrypt.verify(secret, digest)
```
Comment (jhass) [inline spec/std/crypto/bcrypt_spec.cr]: `.should be_true`

**Comment by jhass** [comment]
I think we should keep this open until we fix or drop BCrypt, these vectors are definitely needed.

**Comment by asterite** [comment]
We'll probably remove bcrypt if it's broken, and wait for someone (or us) to implement it well.

**Comment by ysbaddaden** [comment]
I agree with removing BCrypt until someone implements it correctly, or provides a C binding —I could, but there are no reference C implementation, unlike scrypt.

**Comment by jhass** [comment]
Pulled in via #1492, thanks!

**Pull request closed without merging**